### ISSUE TITLE
Respect node settings when rendering

### ISSUE THREAD

**Issue by GuySartorelli** [stefanzweifel/commonmark-markdown-renderer#10]
As discussed in https://github.com/stefanzweifel/commonmark-markdown-renderer/pull/9, various nodes have additional data explaining how they are intended to be rendered. For example, the fenced code node tells you its offset (indent), what delimiter character to use, and how many of that delimiter character should be used.

In some cases respecting this data will be useful only in that you're more likely to match the output to the given input - but in other cases ignoring that data can result in invalid (or at least incorrect) markdown. There's some examples of that in the above linked PR.

I've identified the following as not using the properties available when rendering nodes:
- [x] `StrongRenderer` - ignores `delimiter` #11 
- [x] `EmphasisRenderer` - ignores `delimiter` #11 
- ~~`HtmlBlockRenderer` - ignores `type`~~
- [x] `IndentedCodeRenderer` - doesn't actually render an indented code block - it's rendering a fenced code block - #15
- [x] `ListBlockRenderer` - ignores `isTight`; ignores `delimiter`, `padding`, `bulletChar`, `markerOffset`, `start` from `ListData` - #12
- [x] `ListItemRenderer` - ignores `ListData` (but maybe that's okay, since `ListData` is partially handled in `ListBlockRenderer`?) - #12

**Comment by stefanzweifel** [comment]
This issue seems to be more complex than I thought. 😐 

In #11 I've fixed `StrongRenderer` and `EmphasisRenderer` to use `getOpeningDelimiter()` and `getClosingDelimiter()`.
(For v2 I will probably drop the `use_asterisk` and `use_underscore` config, if that's not something that's comming from commonmark itself; didn't check)

Ignoring `type` in `HtmlBlockRenderer` seems fine to me, as it doesn't seem it should make any impact, on how the node is rendered. (Feel free to drop any examples that teach me otherwise)

`IndentedCodeRenderer` needs redoing. Looking at the code, I now remember that I really struggled to find **exact** examples of what each renderer should look like, when I developed this package.

`ListBlockRenderer` and `ListItemRenderer`: These definitely need an overhaul. In #7 I've fixed an issue with multi-line list items, with a solution that always felt like a cheap fix. 

I need some more head-space to tackle this. And the package definitely needs a better test-suite. The current tests have all been _inspired_ by the league/commonmark suite, but it always felt a bit too unit-testy for me.
Will update the `kitchen-sink.md` with a bit more examples to make testing a bit more straight forward.
